devtools/conda-envs/test_env.yaml

```diff
@@ -22,12 +22,11 @@ dependencies:
   - openff-interchange-base =0.4
   - openff-nagl-base >=0.4.0
   - openff-nagl-models >=0.3.0
+  - typing_extensions
```

Comment:
This is a runtime requirement, not a "only when we run the type-checker" requirement. It's brought in by several other packages so it's not likely to be missed at runtime, but I'm listing it here according to the philosophy of listing all (non-optional) runtime requirements in this sections